add benchmark for quantized batchnorm

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35731 qactivation_benchmarks: small bug fix
* #35729 add more quantized activation benchmarks and input sizes
* **#35389 add benchmark for quantized batchnorm**

Summary:

Adds a benchmark for quantized batchnorm, with the parameters
the same compared to floating point batchnorm benchmark.

Test Plan:

run benchmarks
https://gist.github.com/vkuzo/c49be58abdf0ff64797fab3936d0cb15

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20875543](https://our.internmc.facebook.com/intern/diff/D20875543)

vkuzo committed:
add benchmark for quantized batchnorm

Summary:

Adds a benchmark for quantized batchnorm, with the parameters
the same compared to floating point batchnorm benchmark.

Test Plan:

run benchmarks
https://gist.github.com/vkuzo/c49be58abdf0ff64797fab3936d0cb15

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "add benchmark for quantized batchnorm"


Summary:

Adds a benchmark for quantized batchnorm, with the parameters
the same compared to floating point batchnorm benchmark.

Test Plan:

run benchmarks
https://gist.github.com/vkuzo/c49be58abdf0ff64797fab3936d0cb15

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on benchmarks/operator_benchmark/pt/qbatchnorm_test.py:
```diff
@@ -0,0 +1,53 @@
+    def init(self, M, N, K, device, dtype):
+        # Note: quantized implementation requires rank 4, which is why we
+        # add a 1 as the last dimension
+        self.input_one = torch.rand(M, N, K, 1, device=device, requires_grad=self.auto_set())
```
Can't we instead pass in 4 dimensions (NCHW) to this function?

vkuzo replied:
currently these match the floating point BN benchmark exactly for a fair comparison.  Open to NCHW but would be good to have the fp and q match - we can do this in a separate PR!

supriyar reviewed (approved):
Looks good. Left a small comment.

vkuzo committed:
Update on "add benchmark for quantized batchnorm"


Summary:

Adds a benchmark for quantized batchnorm, with the parameters
the same compared to floating point batchnorm benchmark.

Test Plan:

run benchmarks
https://gist.github.com/vkuzo/c49be58abdf0ff64797fab3936d0cb15

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
